influxdb/influxdb.go

```diff
@@ -2,42 +2,61 @@ package influxdb
 
 import (
 	"context"
+	"log"
 	"os"
 	"time"
 
 	influxdb2 "github.com/influxdata/influxdb-client-go"
 )
 
-func WriteData(serviceName string, containerID string, hostName string, servicePort int, serviceIP string, serviceStatus string, serviceTags []string) {
+type Metrics struct {
+	ServiceName   string
+	ContainerID   string
+	HostName      string
+	ServicePort   int
+	ServiceIP     string
+	ServiceStatus string
+	ServiceTags   []string
+}
+
+///////////////////////
 
-	bucketName := os.Getenv("bucket")
-	influxToken := os.Getenv("influx_token")
-	orgName := os.Getenv("org_name")
-	influxURL := os.Getenv("influx_url")
-	// Create a new client using an InfluxDB server base URL and an authentication token
-	client := influxdb2.NewClient(influxURL, influxToken)
+type InfluxDBClient struct {
+	BucketName  string
+	InfluxToken string
+	InfluxdbURL string
+	client      influxdb2.Client
+}
+
+func New() InfluxDBClient {
+	return InfluxDBClient{
+		BucketName: os.Getenv("bucket"),
+		client:     influxdb2.NewClient(os.Getenv("influx_url"), os.Getenv("influx_token")),
+	}
+}
+
+func (c *InfluxDBClient) WriteData(metrics *Metrics) {
+
+	client := c.client
 	defer client.Close()
 	// Use blocking write client for writes to desired bucket
 
-	writeAPI := client.WriteAPIBlocking(orgName, bucketName)
+	writeAPI := client.WriteAPIBlocking("wkda", c.BucketName)
```

Comment:
I believe the `orgName` should come from the variable and remove the current hard-coded information :)